src/LittleForker/ProcessSupervisor.cs

```diff
@@ -60,27 +61,35 @@ private enum Trigger
         /// <param name="processRunType">
         ///     The process run type.
         /// </param>
+        /// <param name="loggerFactory">
+        ///     A logger factory.
+        /// </param>
         /// <param name="arguments">
         ///     Arguments to be passed to the process.
         /// </param>
         /// <param name="environmentVariables">
         ///     Environment variables that are set before the process starts.
         /// </param>
+        /// <param name="captureStdErr">
+        ///     A flag to indicated whether to capture standard error output.
+        /// </param>
         public ProcessSupervisor(
+            ILoggerFactory loggerFactory,
             ProcessRunType processRunType,
             string workingDirectory,
             string processPath,
             string arguments = null,
             StringDictionary environmentVariables = null,
             bool captureStdErr = false)
         {
+            _loggerFactory = loggerFactory;
             _workingDirectory = workingDirectory;
             _processPath = processPath;
             _arguments = arguments ?? string.Empty;
             _environmentVariables = environmentVariables;
             _captureStdErr = captureStdErr;
 
-            _logger = LogProvider.GetLogger($"ProcessSupervisor-{processPath}");
+            _logger = loggerFactory.CreateLogger($"ProcessSupervisor-{processPath}");
```

Comment:
What do you think about having all the logs outputted from LittleForker to have `SourceContext` starting with `LittleForker.xxxx`. This seems to be the standard also on Microsoft libs and allows us to control better where how the logs are distributed.
In Serilog, for example, we can use:
```csharp
new LoggerConfiguration().Filter.ByExcluding(Matching.FromSource("LittleForker"))
```
Or overriding minimum level
```csharp
new LoggerConfiguration().MinimumLevel.Override("LittleForker", LogEventLevel.Warning)
```


```suggestion
            _logger = loggerFactory.CreateLogger($"LittleForker.ProcessSupervisor-{processPath}");
```